Rework procedure to enable BMC on Smart Proxies (#2825)
theforeman/foreman-documentation: guides/common/modules/proc_adding-a-bmc-interface.adoc

Author: The way I understand this: foreman-installer installs a package that has a dependency to "ipmitools"; so we don't have to document it/install it manually. See https://github.com/theforeman/foreman-documentation/pull/2825#discussion_r1521888041

Is this correct for Foreman nightly? @ekohl

Reviewer: @maximiliankolb it should be fixed for RPMs in nightly. For 3.8, 3.9 and 3.10 it'll need to wait for the next release. I also noticed it's a problem for Debian, so it may not be a bad thing to keep it in for now.

Author: Thank you; confirmed locally:

````
# run AlmaLinux 8.9
# add Foreman nightly repo

# dnf install rubygem-rubyipmi
Installing:
 rubygem-rubyipmi                                   noarch                               0.11.1-2.el8                                                           foreman                                  35 k
Installing dependencies:
 ipmitool                                           x86_64                               1.8.18-19.el8                                                          appstream                               394 k
 ruby        
````

Before, I was searching for "ipmitool" on yum.theforeman.org and couldn't find it. Well d'oh! So yes, we do not need to mention it in the docs for nightly.

So we should rebase on merge and then only cherry-pick the first two commits to "3.10" and "3.9".

[id="Adding_a_Baseboard_Management_Controller_Interface_{context}"]
= Adding a baseboard management controller (BMC) interface

> {Project} can control the power status of bare metal hosts.

Reviewer 2: ```suggestion
{Project} can control the power status of bare-metal hosts.
```

Author: Applied and created a commit to fix all other instances.

Use this procedure to configure a baseboard management controller (BMC) interface for a host that supports this feature.

.Prerequisites

> * You know the MAC address, IP address, and other details of the BMC interface on the host, and the appropriate credentials for that interface.

Reviewer 2: ```suggestion
* You know the MAC address, IP address, and other details of the BMC interface on the host, and authentication credentials for that interface.
```
Would this make it clearer what the appropriate credentials are? (Non blocker)

Author: Yes, applied.

[id="Adding_a_Baseboard_Management_Controller_Interface_{context}"]
= Adding a baseboard management controller (BMC) interface

You can control the power status of bare-metal hosts from {Project}.
Use this procedure to configure a baseboard management controller (BMC) interface for a host that supports this feature.

.Prerequisites
* You know the MAC address, IP address, and other details of the BMC interface on the host, and authentication credentials for that interface.
+
[NOTE]
====
You only need the MAC address for the BMC interface if the BMC interface is managed, so that it can create a DHCP reservation.
====

.Procedure
. Enable BMC power management on your {SmartProxy}:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----
# {foreman-installer} \
--foreman-proxy-bmc-default-provider=ipmitool \
--foreman-proxy-bmc=true
----
. In the {ProjectWebUI}, navigate to *Infrastructure* > *Subnets*.
. Select the subnet of your host.
ifdef::satellite[]
. On the *{SmartProxies}* tab, select your {SmartProxy} as *BMC {SmartProxy}*.
endif::[]
ifndef::satellite[]
. On the *Proxies* tab, select your {SmartProxy} as *BMC Proxy*.

> endif::[]

Reviewer 2: Just FYI, I'm confirming the branding for Satellite. This is a superb solution, btw!

Author: I've verified it already :see_no_evil: :)

Reviewer 2: All good. :)

[id="Adding_a_Baseboard_Management_Controller_Interface_{context}"]
= Adding a baseboard management controller (BMC) interface

> {Project} can control the power status of bare-metal hosts.

Reviewer 2: ```suggestion
You can control the power status of bare-metal hosts from {Project}.
```

Use this procedure to configure a baseboard management controller (BMC) interface for a host that supports this feature.

.Prerequisites
* You know the MAC address, IP address, and other details of the BMC interface on the host, and authentication credentials for that interface.
+
[NOTE]
====
You only need the MAC address for the BMC interface if the BMC interface is managed, so that it can create a DHCP reservation.
====

.Procedure
. Install `ipmitool` on your {SmartProxy}:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----

> # {project-package-install} ipmitool

Reviewer: This shouldn't be needed. It looks like this is a regression introduced in https://github.com/theforeman/foreman-packaging/commit/6cc3c98edd01fe2f24db8a426b9eed405c283200 so I've opened https://github.com/theforeman/foreman-packaging/pull/10556 instead. Interestingly enough, it looks like it was never present in Debian packages.

----
. Enable BMC power management on your {SmartProxy}:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----
# {foreman-installer} \
--foreman-proxy-bmc-default-provider=ipmitool \

> --foreman-proxy-bmc=true

Reviewer: I think the old order is more logical: first you enable it, then you select the default provider.

Reviewer 2: I agree. I would generally prefer to sort the arguments by relevance instead of alphabetically.

> # {foreman-installer} \
> --foreman-proxy-bmc-default-provider=ipmitool \
> --foreman-proxy-bmc=true

Reviewer 2: ```suggestion
# {foreman-installer} \
--foreman-proxy-bmc-default-provider ipmitool \
--foreman-proxy-bmc true
```
I'm not sure of the doc preference for listing arguments (space vs. =). Functionally, they are the same. (non blocker)

Author: I have created an issue for this. Let's decide on one solution and then fix all instances: https://github.com/theforeman/foreman-documentation/issues/2881

----
. In the {ProjectWebUI}, navigate to *Infrastructure* > *{SmartProxies}*.

> . Optional: From the list in the *Actions* column, select *Refresh*.

Reviewer: I'd like to avoid telling users to refresh. If anything, flip it around: Verify BMC is present, otherwise refresh.

Author: Dropped!

====

.Procedure

> . Install `ipmitool` on your {SmartProxy}:

Reviewer 3: I am not sure why you moved the ipmitool installation to the procedure. I would have kept it in the prerequisites. The installation process is fairly simple, so I think the user could figure out how to install the tool without the command.

Reviewer 2: In other places, like the Provisioning Hosts guide, we include package installations in the procedures. I don't mind moving it.

Author: I am OK either way. I lean towards "procedure" because it feel likes "just another step" to do.

+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----
# {project-package-install} ipmitool
----
. Enable BMC power management on your {SmartProxy}:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----
# {foreman-installer} \
--foreman-proxy-bmc-default-provider=ipmitool \
--foreman-proxy-bmc=true
----
. In the {ProjectWebUI}, navigate to *Infrastructure* > *{SmartProxies}*.

> . From the list in the *Actions* column, click *Refresh*.

Reviewer 2: ```suggestion
. Optional: From the list in the *Actions* column, click *Refresh*.
```
This shouldn't be necessary.

Author: Applied, but maybe we should drop it completely. I believe Ewoud commented on another PR of mine saying that `foreman-installer` takes care of this.

Reviewer 2: Okay, let's drop it.

> . From the list in the *Actions* column, click *Refresh*.

Author: Applied.

Ensure that the list of features includes *BMC*.
. Navigate to *Infrastructure* > *Subnets*.
. Select the subnet of your host.

> . On the *Proxies* tab, select your {SmartProxy} as *BMC Proxy*.

Reviewer 2: ```suggestion
. On the *{SmartProxies}* tab, select your {SmartProxy} as *BMC {SmartProxy}*.
```
Unfortunately, these are branded. I need to double check.

Author: I've checked & applied branding. I think I can live with this too.

Reviewer 4: Is this rather a Satellite vs. Non-Satellite thing?

Author: Yes, fixed.
